### PR TITLE
Improves tgui build speed

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
Swaps out our minimizer, Terser, with esbuildplugin.  This is the engine behind vite. This is a pretty straight forward replacement, let's let the numbers do the talking:

<details>
<summary>Test environment / My specs</summary>

intel i7-13700kf 
32GB RAM
windows 11 
intel 660p m.2 SSD
asus 4080 TUF series

</details>

<details>
<summary>Fresh build</summary>

Test steps:

1. Delete `tgui/.yarn/webpack` folder
2. `CTRL SHIFT B`

Results:
fresh

terser averages `25.16s`
esbuild averages `19.37s` - 23% faster

</details>

<details>
<summary>Cached build</summary>

(assuming files are already there, CTRL SHIFT B if not)

1. add a comment to an interface file
2. `CTRL SHIFT B`

Results:
cached

terser averages `9s`
esbuild averages `3.52s` - 60.91% faster

</details>

## Why It's Good For The Game
Faster dev environment
## Changelog
N/A nothing player facing

### CODE REVIEW

File: tgui/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "private": true,
   "name": "tgui-workspace",
-  "version": "4.3.1",
+  "version": "4.3.2",
```
Comment (ZephyrTFA) [inline tgui/package.json:4]: this should probably be a minor version bump not a build version bump

Comment (jlsnow301) [reply tgui/package.json:4]: sry these were from #79895

the ~only~ change here is the one in webpack.config.js

Comment (jlsnow301) [reply tgui/package.json:4]: (but I agree)

**Comment by MrStonedOne** [comment]
Works on windows 7, but I noticed the inner node executable hangs around for 10 seconds with esbuild.exe running after the build task ends. are you sure your savings are just from it releasing control early?

On production `tgstation-server` does need to know when it can safely trust the compile process has fully completed, this won't be  an issue right away because tgstation-server handles building byond, not juke, so there will be a second compilation step anyways that will keep cleanup from happening too early, but it is something to look into.

I did however notice that peak memory usage is lower, which will help with compilation on prod where we have tighter memory budgets for vm-all-the-things reasons.

**Comment by jlsnow301** [comment]
Drafting this as #79916 seems to be the superior option. If that PR doesn't work out, this is a good compromise.

**Review by MrsTonedOne — APPROVED** [review]

**Pull request merged**